test/mock/grpc/mock_management_plane_grpc_file_server.go

```diff
@@ -0,0 +1,268 @@
+
+		err := performFileAction(fileAction, fileContents, fullFilePath, filePermissions)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &v1.UpdateFileResponse{
+		FileMeta: fileMeta,
+	}, nil
+}
+
+func (mgfs *ManagementGrpcFileServer) getConfigVersions(fileName, fileHash string) []string {
+	fileConfigVersions := []string{}
+
+	for configVersion, overview := range mgfs.overviews {
+		for _, file := range overview {
+			if fileName == file.GetFileMeta().GetName() && fileHash == file.GetFileMeta().GetHash() {
+				fileConfigVersions = append(fileConfigVersions, configVersion)
+				break
+			}
+		}
+	}
+
+	return fileConfigVersions
+}
+
+func getMapOfVersionedFiles(configDirectory string) (map[string][]*v1.File, error) {
+	files := make(map[string][]*v1.File)
+
+	err := filepath.Walk(configDirectory, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			// nolint: gomnd
+			splitPath := strings.SplitN(strings.Split(path, configDirectory)[1], "/", 3)
+			version := splitPath[1]
+			filePath := "/" + splitPath[2]
+			versionDirectory := filepath.Join(configDirectory, version)
+
+			hash, err := getFileHash(path)
+			if err != nil {
+				return err
+			}
+
+			fileInfo, err := os.Stat(path)
+			if err != nil {
+				return err
+			}
+
+			files[versionDirectory] = append(files[versionDirectory], &v1.File{
+				FileMeta: &v1.FileMeta{
+					Name:         filePath,
+					Hash:         hash,
+					ModifiedTime: timestamppb.New(fileInfo.ModTime()),
+					Permissions:  fileInfo.Mode().Perm().String(),
+					Size:         fileInfo.Size(),
+				},
+			})
+		}
+
+		return nil
+	})
+
+	return files, err
+}
+
+func generateConfigVersion(files []*v1.File) string {
+	var hashes string
+
+	for _, file := range files {
+		hashes += file.GetFileMeta().GetHash()
+	}
+
+	return uuid.Generate("%s", hashes)
+}
+
+func getFileHash(filePath string) (string, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		return "", err
+	}
+
+	return string(h.Sum(nil)), nil
+}
+
+func performFileAction(fileAction v1.File_FileAction, fileContents []byte, fullFilePath, filePermissions string) error {
+	if fileAction == v1.File_FILE_ACTION_ADD {
+		err := os.WriteFile(fullFilePath, fileContents, getFileMode(filePermissions))
+		if err != nil {
+			slog.Info("Failed to create/update file", "full_file_path", fullFilePath, "error", err)
+			return status.Errorf(codes.Internal, "Failed to create/update file")
+		}
+	} else if fileAction == v1.File_FILE_ACTION_DELETE {
+		err := os.Remove(fullFilePath)
+		if err != nil {
+			slog.Info("Failed to delete file", "full_file_path", fullFilePath, "error", err)
+			return status.Errorf(codes.Internal, "Failed to delete file")
+		}
+	} else {
+		slog.Info("Nothing to update, file action is unchanged", "full_file_path", fullFilePath)
+	}
+
+	return nil
```

Comment:
I suppose this will need to be a `switch` at one point anyway, so should we just add it now?
```suggestion
	switch fileAction {
	case v1.File_FILE_ACTION_ADD:
		err := os.WriteFile(fullFilePath, fileContents, getFileMode(filePermissions))
		if err != nil {
			slog.Info("Failed to create/update file", "full_file_path", fullFilePath, "error", err)
			return status.Errorf(codes.Internal, "Failed to create/update file")
		}
	case v1.File_FILE_ACTION_DELETE:
		err := os.Remove(fullFilePath)
		if err != nil {
			slog.Info("Failed to delete file", "full_file_path", fullFilePath, "error", err)
			return status.Errorf(codes.Internal, "Failed to delete file")
		}
	case v1.File_FILE_ACTION_UPDATE:
		fallthrough
	case v1.File_FILE_ACTION_UNSPECIFIED:
		fallthrough
	case v1.File_FILE_ACTION_UNCHANGED:
		fallthrough
	default:
		slog.Info("Nothing to update, file action is unchanged", "full_file_path", fullFilePath)
	}

	return nil
```
Unless we plan on not adding the other file actions to the mock server 🤔

Reply:
Ah I forgot about the other file actions. Let me update that.